airbyte-integrations/connectors/source-mongodb-v2/src/main/java/io/airbyte/integrations/source/mongodb/MongoUtil.java

```diff
@@ -206,6 +210,40 @@ public static Optional<CollectionStatistics> getCollectionStatistics(final Mongo
     return Optional.empty();
   }
 
+  public static int getChunkSizeForCollection(final Optional<CollectionStatistics> collectionStatistics, final ConfiguredAirbyteStream stream) {
+    // If table size info could not be calculated, a default chunk size will be provided.
+    if (collectionStatistics.isEmpty() || shouldUseDefaultChunkSize(collectionStatistics.get())) {
+      LOGGER.info("Chunk size could not be determined for: {}.{}, defaulting to {} rows", stream.getStream().getNamespace(),
+          stream.getStream().getName(), DEFAULT_CHUNK_SIZE);
+      return DEFAULT_CHUNK_SIZE;
+    }
+    CollectionStatistics stats = collectionStatistics.get();
+    final long totalRows = stats.count().longValue();
+    final long totalBytes = stats.size().longValue();
```

Comment:
Is this a relatively cheap query?

Reply:
Yes, this is just querying the metadata (collStats). Unlike Postgres/other relational dbsn this isn't very expensive

Reply:
Benchmarked on our test data set - no real difference in timing

Reply:
Great! I'm sensitive because over in destinations-land, doing a similar information_schema query for Snowflake is expensive